### ISSUE TITLE
snap fails to process with "bin/desktop-launch $SNAP/app/my-app does not exist lint-snap-v2_command"

### ISSUE THREAD

**Issue by kspearrin** [electron-userland/electron-builder#2614]
* **Version**: 20.0.7

* **Target**: `build --linux --x64`

Now that https://github.com/electron-userland/electron-builder/issues/2596 has been resolved I have tried pushing a packaged `.snap` for our app again.

The snap successfully uploads this time, yet it now fails the validation process with the following:

```
1 Warning:
unknown fields for app 'my-app': 'adapter' lint-snap-v2_apps_unknown (my-app) 
```

```
1 Fail:
bin/desktop-launch $SNAP/app/my-app does not exist lint-snap-v2_command (my-app) 
```

and 60 passes....

**Comment by develar** [comment]
Workaround: set snap.useTemplateApp to false

```
"snap": {
      "useTemplateApp": false,
    },
```

In this case you can build snap only on Linux (better to use Ubuntu 16 LTS) or using Docker for macOS. And build time will be 2-10 minutes.

**Comment by Qiplex** [comment]
Thanks @develar !

**Comment by Qiplex** [comment]
@develar here is outcome I've got with: 

electron-builder: 2.0.8
os: ubuntu 17.10

```
$ time electron-builder -l snap

  • electron-builder version=20.0.8
  • loaded configuration file=package.json ("build" field)
  • writing effective config file=release-builds/electron-builder.yaml
  • rebuilding native production dependencies platform=linux arch=x64
  • packaging       platform=linux arch=x64 electron=1.8.2 appOutDir=release-builds/linux-unpacked
  • building        target=snap arch=x64 file=release-builds/easy-disk-cleaner_1.0.2_amd64.snap
  • application Linux category is set to default "Utility" reason=linux.category is not set and cannot map from macOS docs=https://electron.build/configuration/configuration#LinuxBuildOptions-category
Setting target machine to 'amd64'
Preparing to pull desktop-gtk2 
...................................................................................................................

Pulling app 
Preparing to build app 
Building app 
Staging app 
Priming desktop-gtk2 
Files from the build host were migrated into the snap to satisfy dependencies that would otherwise not be met. This feature will be removed in a future release. If these libraries are needed in the final snap, ensure that the following are either satisfied by a stage-packages entry or through a part:
lib/x86_64-linux-gnu/libblkid.so.1
lib/x86_64-linux-gnu/libcom_err.so.2
lib/x86_64-linux-gnu/libgcc_s.so.1
lib/x86_64-linux-gnu/libgcrypt.so.20
lib/x86_64-linux-gnu/libgpg-error.so.0
lib/x86_64-linux-gnu/liblzma.so.5
lib/x86_64-linux-gnu/libmount.so.1
lib/x86_64-linux-gnu/libselinux.so.1
lib/x86_64-linux-gnu/libsystemd.so.0
lib/x86_64-linux-gnu/libuuid.so.1
lib/x86_64-linux-gnu/libz.so.1
usr/lib/x86_64-linux-gnu/libstdc++.so.6
Traceback (most recent call last):
  File "/usr/local/bin/snapcraft", line 7, in <module>
    from snapcraft.cli.__main__ import run
  File "/usr/lib/python3/dist-packages/snapcraft/cli/__main__.py", line 43, in <module>
    run(prog_name='snapcraft')
  File "/usr/lib/python3/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/snapcraft/cli/lifecycle.py", line 111, in prime
    _execute('prime', parts, **kwargs)
  File "/usr/lib/python3/dist-packages/snapcraft/cli/lifecycle.py", line 33, in _execute
    lifecycle.execute(command, project_options, parts)
  File "/usr/lib/python3/dist-packages/snapcraft/internal/lifecycle/_runner.py", line 79, in execute
    _Executor(config, project_options).run(step, part_names)
  File "/usr/lib/python3/dist-packages/snapcraft/internal/lifecycle/_runner.py", line 185, in run
    self._run_step(step, part, part_names)
  File "/usr/lib/python3/dist-packages/snapcraft/internal/lifecycle/_runner.py", line 222, in _run_step
    getattr(part, step)()
  File "/usr/lib/python3/dist-packages/snapcraft/internal/pluginhandler/__init__.py", line 589, in prime
    elf_patcher.patch(elf_file=elf_file)
  File "/usr/lib/python3/dist-packages/snapcraft/internal/elf.py", line 392, in patch
    elf_file_path=elf_file.path)
  File "/usr/lib/python3/dist-packages/snapcraft/internal/elf.py", line 297, in wrapper
    return f(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/snapcraft/internal/elf.py", line 407, in _run_patchelf
    subprocess.check_call(cmd)
  File "/usr/lib/python3.6/subprocess.py", line 286, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/usr/lib/python3.6/subprocess.py", line 267, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/usr/lib/python3.6/subprocess.py", line 709, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.6/subprocess.py", line 1275, in _execute_child
    restore_signals, start_new_session, preexec_fn)
TypeError: expected str, bytes or os.PathLike object, not NoneType
  ⨯ exit status 1
output: Traceback (most recent call last):
  File "/usr/local/bin/snapcraft", line 7, in <module>
    from snapcraft.cli.__main__ import run
  File "/usr/lib/python3/dist-packages/snapcraft/cli/__main__.py", line 43, in <module>
    run(prog_name='snapcraft')
  File "/usr/lib/python3/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/snapcraft/cli/lifecycle.py", line 158, in pack
    snap_name = lifecycle.pack(directory, output)
  File "/usr/lib/python3/dist-packages/snapcraft/internal/lifecycle/_packer.py", line 57, in pack
    snap = _snap_data_from_dir(directory)
  File "/usr/lib/python3/dist-packages/snapcraft/internal/lifecycle/_packer.py", line 34, in _snap_data_from_dir
    with open(os.path.join(directory, 'meta', 'snap.yaml')) as f:
FileNotFoundError: [Errno 2] No such file or directory: '/home/osboxes/edc/release-builds/__snap-x64/prime/meta/snap.yaml'


Error: /usr/local/lib/node_modules/electron-builder/node_modules/app-builder-bin-linux/x64/app-builder exited with code 1
    at ChildProcess.childProcess.once.code (/usr/local/lib/node_modules/electron-builder/node_modules/builder-util/src/util.ts:252:14)
    at ChildProcess.g (events.js:292:16)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:920:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:230:5)
From previous event:
    at SnapTarget.build (/usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/out/targets/snap.js:174:11)
    at taskManager.addTask.default.map.it (/usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/src/platformPackager.ts:121:67)
From previous event:
    at LinuxPackager.packageInDistributableFormat (/usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/src/platformPackager.ts:121:23)
    at /usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/src/platformPackager.ts:116:10
    at next (native)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
From previous event:
    at LinuxPackager.pack (/usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/out/platformPackager.js:188:11)
    at /usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/src/packager.ts:372:24
    at next (native)
    at xfs.stat (/usr/local/lib/node_modules/electron-builder/node_modules/fs-extra/lib/mkdirs/mkdirs.js:56:16)
    at /usr/local/lib/node_modules/electron-builder/node_modules/graceful-fs/polyfills.js:287:18
From previous event:
    at Packager.doBuild (/usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/out/packager.js:441:11)
    at /usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/src/packager.ts:316:52
    at next (native)
    at /usr/local/lib/node_modules/electron-builder/node_modules/graceful-fs/graceful-fs.js:99:16
    at /usr/local/lib/node_modules/electron-builder/node_modules/graceful-fs/graceful-fs.js:43:10
    at FSReqWrap.oncomplete (fs.js:123:15)
From previous event:
    at Packager._build (/usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/out/packager.js:385:11)
    at /usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/src/packager.ts:278:23
    at next (native)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
From previous event:
    at Packager.build (/usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/out/packager.js:341:11)
    at /usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/src/index.ts:50:40
    at next (native)
From previous event:
    at build (/usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/out/index.js:47:21)
    at build (/usr/local/lib/node_modules/electron-builder/src/builder.ts:228:10)
    at then (/usr/local/lib/node_modules/electron-builder/src/cli/cli.ts:48:33)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
From previous event:
    at Object.args [as handler] (/usr/local/lib/node_modules/electron-builder/src/cli/cli.ts:48:33)
    at Object.runCommand (/usr/local/lib/node_modules/electron-builder/node_modules/yargs/lib/command.js:235:44)
    at Object.parseArgs [as _parseArgs] (/usr/local/lib/node_modules/electron-builder/node_modules/yargs/yargs.js:1042:24)
    at Object.get [as argv] (/usr/local/lib/node_modules/electron-builder/node_modules/yargs/yargs.js:957:21)
    at Object.<anonymous> (/usr/local/lib/node_modules/electron-builder/src/cli/cli.ts:42:15)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:383:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:496:3

real	19m29.489s
user	9m18.924s
sys	0m20.366s

```

```
osboxes@osboxes:~/edc$ ls -la /home/osboxes/edc/release-builds/__snap-x64/prime/meta/snap.yaml
ls: cannot access '/home/osboxes/edc/release-builds/__snap-x64/prime/meta/snap.yaml': No such file or directory

osboxes@osboxes:~/edc$ find /home/osboxes/edc/release-builds/__snap-x64/ -name snap.yaml
osboxes@osboxes:~/edc$
```

**Comment by Qiplex** [comment]
@develar I just gave it one more try on Mac with Docker. See outcome below:

electron-builder: 20.0.8
os: macOS Sierra 10.12.6

```
Issues while validating snapcraft.yaml: The 'apps/easy-disk-cleaner' property does not match the required schema: Additional properties are not allowed ('adapter' was unexpected)
  ⨯ exit status 2
```
```
 /opt/bin/edc   master ●  time electron-builder -l snap

  • electron-builder version=20.0.8
  • loaded configuration file=package.json ("build" field)
  • writing effective config file=release-builds/electron-builder.yaml
  • rebuilding native production dependencies platform=linux arch=x64
  • packaging       platform=linux arch=x64 electron=1.7.9 appOutDir=release-builds/linux-unpacked
  • building        target=snap arch=x64 file=release-builds/easy-disk-cleaner_1.0.2_amd64.snap
  • application Linux category is set to default "Utility" reason=linux.category is not set and cannot map from macOS docs=https://electron.build/configuration/configuration#LinuxBuildOptions-category
Setting target machine to 'amd64'
Issues while validating snapcraft.yaml: The 'apps/easy-disk-cleaner' property does not match the required schema: Additional properties are not allowed ('adapter' was unexpected)
  ⨯ exit status 2
github.com/develar/app-builder/pkg/util.ExecuteWithInheritedStdOutAndStdErr
	/Users/develar/go/src/github.com/develar/app-builder/pkg/util/util.go:69
github.com/develar/app-builder/pkg/snap.buildUsingDocker
	/Users/develar/go/src/github.com/develar/app-builder/pkg/snap/snap.go:232
github.com/develar/app-builder/pkg/snap.Snap
	/Users/develar/go/src/github.com/develar/app-builder/pkg/snap/snap.go:170
github.com/develar/app-builder/pkg/snap.ConfigureCommand.func1
	/Users/develar/go/src/github.com/develar/app-builder/pkg/snap/snap.go:90
github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin.(*actionMixin).applyActions
	/Users/develar/go/src/github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin/actions.go:28
github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin.(*Application).applyActions
	/Users/develar/go/src/github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin/app.go:557
github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin.(*Application).execute
	/Users/develar/go/src/github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin/app.go:390
github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin.(*Application).Parse
	/Users/develar/go/src/github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin/app.go:222
main.main
	/Users/develar/go/src/github.com/develar/app-builder/main.go:59
runtime.main
	/usr/local/Cellar/go/1.10/libexec/src/runtime/proc.go:198
runtime.goexit
	/usr/local/Cellar/go/1.10/libexec/src/runtime/asm_amd64.s:2361

Error: /usr/local/lib/node_modules/electron-builder/node_modules/app-builder-bin-mac/app-builder exited with code 1
    at ChildProcess.childProcess.once.code (/usr/local/lib/node_modules/electron-builder/node_modules/builder-util/src/util.ts:252:14)
    at Object.onceWrapper (events.js:318:30)
    at emitTwo (events.js:125:13)
    at ChildProcess.emit (events.js:213:7)
    at maybeClose (internal/child_process.js:921:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
From previous event:
    at SnapTarget.build (/usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/out/targets/snap.js:174:11)
    at taskManager.addTask.default.map.it (/usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/src/platformPackager.ts:121:67)
From previous event:
    at LinuxPackager.packageInDistributableFormat (/usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/src/platformPackager.ts:121:23)
    at /usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/src/platformPackager.ts:116:10
    at Generator.next (<anonymous>)
    at runCallback (timers.js:781:20)
    at tryOnImmediate (timers.js:743:5)
    at processImmediate [as _immediateCallback] (timers.js:714:5)
From previous event:
    at LinuxPackager.pack (/usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/out/platformPackager.js:188:11)
    at /usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/src/packager.ts:372:24
    at Generator.next (<anonymous>)
    at xfs.stat (/usr/local/lib/node_modules/electron-builder/node_modules/fs-extra/lib/mkdirs/mkdirs.js:56:16)
    at /usr/local/lib/node_modules/electron-builder/node_modules/graceful-fs/polyfills.js:287:18
    at FSReqWrap.oncomplete (fs.js:153:5)
From previous event:
    at Packager.doBuild (/usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/out/packager.js:441:11)
    at /usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/src/packager.ts:316:52
    at Generator.next (<anonymous>)
    at /usr/local/lib/node_modules/electron-builder/node_modules/graceful-fs/graceful-fs.js:99:16
    at /usr/local/lib/node_modules/electron-builder/node_modules/graceful-fs/graceful-fs.js:43:10
    at FSReqWrap.oncomplete (fs.js:135:15)
From previous event:
    at Packager._build (/usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/out/packager.js:385:11)
    at /usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/src/packager.ts:278:23
    at Generator.next (<anonymous>)
    at runCallback (timers.js:781:20)
    at tryOnImmediate (timers.js:743:5)
    at processImmediate [as _immediateCallback] (timers.js:714:5)
From previous event:
    at Packager.build (/usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/out/packager.js:341:11)
    at /usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/src/index.ts:50:40
    at Generator.next (<anonymous>)
From previous event:
    at build (/usr/local/lib/node_modules/electron-builder/node_modules/electron-builder-lib/out/index.js:47:21)
    at build (/usr/local/lib/node_modules/electron-builder/src/builder.ts:228:10)
    at then (/usr/local/lib/node_modules/electron-builder/src/cli/cli.ts:48:33)
    at runCallback (timers.js:781:20)
    at tryOnImmediate (timers.js:743:5)
    at processImmediate [as _immediateCallback] (timers.js:714:5)
From previous event:
    at Object.args [as handler] (/usr/local/lib/node_modules/electron-builder/out/cli/cli.js:121:117)
    at Object.runCommand (/usr/local/lib/node_modules/electron-builder/node_modules/yargs/lib/command.js:235:44)
    at Object.parseArgs [as _parseArgs] (/usr/local/lib/node_modules/electron-builder/node_modules/yargs/yargs.js:1042:24)
    at Object.get [as argv] (/usr/local/lib/node_modules/electron-builder/node_modules/yargs/yargs.js:957:21)
    at Object.<anonymous> (/usr/local/lib/node_modules/electron-builder/out/cli/cli.js:117:439)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Function.Module.runMain (module.js:605:10)
    at startup (bootstrap_node.js:158:16)
    at bootstrap_node.js:575:3

electron-builder -l snap  15.66s user 1.84s system 88% cpu 19.763 total
```

snap config was the same as on Ubuntu VM

```
"snap": {
      "confinement": "classic",
      "useTemplateApp": "false"
    },

```

**Comment by jdstrand** [comment]
The 'adapter' denial is a bug in snapcraft: https://bugs.launchpad.net/snapcraft/+bug/1750658. It looks like @kalikiana started to look at it.

**Comment by jdstrand** [comment]
@kspearrin - please see my last comment for 'adapter'.

The issue with `bin/desktop-launch $SNAP/app/my-app does not exist lint-snap-v2_command (my-app)` is a bug in our review tools, which I'll fix. If you request a manual review, I'll approve it (though see below).

If you remove `adapter: none` from your snapcraft.yaml, this will pass again. What is happening is that with `adapter: none` a command wrapper is used in the resulting meta/snap.yaml such that it has `command: command-tusk.wrapper`, which passes review. With the new `adapter: none` yaml, snapcraft passes the `command` unmodified from snapcraft.yaml to meta/snap.yaml (ie `bin/desktop-launch $SNAP/app/my-app`) and the review tools aren't splitting on spaces and so it treats the command as `bin/desktop-launch $SNAP/app/my-app` when it should be `bin/desktop-launch`.

**Comment by jdstrand** [comment]
@kspearrin - as an aside, fyi your snap has many directories that have '777' permissions. This isn't a problem per se, but may not be what you intended.

**Comment by jdstrand** [comment]
Oh, and if I wasn't clear, IMO there is no bug in electron-builder for the reported issue.

**Comment by kspearrin** [comment]
@jdstrand I'll try another submission later today with `"useTemplateApp": "false"`. 

> @kspearrin - as an aside, fyi your snap has many directories that have '777' permissions. This isn't a problem per se, but may not be what you intended.

I don't create any directories in my app. What directories are you referring to?

**Comment by jdstrand** [comment]
Actually, it is more than just dirs. Eg:
```
$ unsquashfs -lls tusk_8.snap|grep 'rwxrwxrwx' |grep -v 'lrwxrwxrwx # symlinks should be 777, so don't list them
drwxrwxrwx root/root               138 2018-02-22 06:54 squashfs-root
drwxrwxrwx root/root               440 2018-02-22 06:54 squashfs-root/app
-rwxrwxrwx root/root              1060 2018-02-06 14:07 squashfs-root/app/LICENSE.electron.txt
-rwxrwxrwx root/root           1789407 2018-02-06 14:09 squashfs-root/app/LICENSES.chromium.html
-rwxrwxrwx root/root             25856 2018-02-06 14:11 squashfs-root/app/blink_image_resources_200_percent.pak
...
```

This might be a result of the filesystem you are packing the snap on. I've seen, for example, that sharing files between a MacOS host and Ubuntu guest via Parallels and having the files on the Parallels fs do strange things (I'm not a Parallels user and this may be fixed now). If you are on an 'exotic' fs, perhaps try moving the files to a POSIX-compliant fs (eg, ext4) and see if that makes a difference.

**Comment by kspearrin** [comment]
The snap is being compiled on an Ubuntu 16.04 VM.

**Comment by jdstrand** [comment]
Are you doing file sharing between the host and the VM? If so, is your build happening on the shared fs? If so, what happens if you do your built in the VM outside of the shared area?

If not, what is the output of `umask` in your VM?

**Comment by kspearrin** [comment]
This is an "almost fresh" install of Ubuntu 16.04 via Hyper-V on Windows 10. I just installed it the other day and have only been using it for electron packaging so far. There is no sharing or anything like that with the host. I am packaging from `~/Projects/myapp`.

```
$ umask
0002
```

**Comment by develar** [comment]
@jdstrand Thanks for details about `adapter`. It is also bug of electron-builder for prepacked snap — now `adapter` will be not generated in this case. For full building, I will add temp fix (until https://bugs.launchpad.net/snapcraft/+bug/1750658 is not foxed) to remove `adapter` from generated `snap.yaml`.

**Comment by jdstrand** [comment]
Ok, then it shouldn't be anything you are doing. I'm not familiar with electron-builder. Perhaps it is changing the `umask` internally? Perhaps it downloads an archive file and unpacks it, and the files in the archive have the wrong permissions? I recommend filing a different issue and see if an electron-builder developer can comment.

**Comment by kspearrin** [comment]
I opened https://github.com/electron-userland/electron-builder/issues/2627

**Comment by kspearrin** [comment]
@develar I added

```
"snap": {
      "useTemplateApp": false,
    },
```

Yet I still get the same errors when trying to `snapcraft push`.

**Comment by develar** [comment]
`adapter: none` issue is fixed in the 20.0.9 (for both cases — prepacked snap or full build).

file mode issue will be fixed tomorrow morning CET.
